lib/exgboost/parameters.ex

```diff
@@ -0,0 +1,968 @@
+      `:silent`, `:warning`, `:info`, `:debug`.
+      """
+    ],
+    use_rmm: [
+      type: :boolean,
+      default: false,
+      doc: """
+      Whether to use RAPIDS Memory Manager for memory allocation.
+      This This option is only applicable when XGBoost is built (compiled)
+      with the RMM plugin enabled. Valid values are `true` and `false`.
+      """
+    ]
+  ]
+
+  @general_params [
+    booster: [
+      type: {:in, [:gbtree, :gblinear, :dart]},
+      default: :gbtree,
+      doc: """
+      Which booster to use. Valid values are `:gbtree`, `:gblinear`, `:dart`
+          * `:gbtree`: tree-based models
+          * `:gblinear`: linear models
+          * `:dart`: tree-based models with dropouts
+      """
+    ],
+    verbosity: [
+      type: {:custom, EXGBoost.Parameters, :validate_verbosity, []},
+      default: :silent,
+      doc: """
+      Verbosity of printing messages. Valid values are:
+      `:silent`, `:warning`, `:info`, `:debug`
+      """
+    ],
+    validate_parameters: [
+      type: :boolean,
+      default: true,
+      doc: """
+      Whether to perform validation of parameters. If set to `true`, an error
+      will be raised if an invalid parameter is passed to the booster, and
+      EXGBoost will take care of formatting all parameters to the expected
+      input of XGBoost. If set to `false`, the user is responsible for ensuring
+      that all parameters are valid strings which is what XGBoost is expecting.
+      """
+    ],
+    nthread: [
+      type: :non_neg_integer,
+      default: 0,
+      doc: """
+      Number of threads to use for training and prediction. If `0`, then the
+      number of threads is set to the number of cores.
+      """
+    ],
+    disable_default_eval_metric: [
+      type: :boolean,
+      default: false,
+      doc: """
+      Whether to disable the default metric. If set to `true`, then the default
+      metric is not used for evaluation. This is useful when using custom
+      evaluation metrics.
+      """
+    ],
+    num_features: [
+      type: :non_neg_integer,
+      doc: """
+      Feature dimension used in boosting, set to maximum dimension of the feature
+      """
+    ]
+  ]
+
+  @tree_booster_params [
+    eta: [
+      type: {:custom, EXGBoost.Parameters, :in_range, ["[0,1]"]},
+      default: 0.3,
+      doc: """
+      Step size shrinkage used in update to prevents overfitting. After each
+      boosting step, we can directly get the weights of new features. and `eta`
+      actually shrinks the feature weights to make the boosting process more
+      conservative. Valid range is `[0,1]`.
+      """
+    ],
+    gamma: [
+      type: {:custom, EXGBoost.Parameters, :in_range, ["[0,:inf]"]},
+      default: 0.0,
+      doc: ~S'''
+      Minimum loss reduction required to make a further partition on a leaf node
+      of the tree. The larger `gamma` is, the more conservative the algorithm will
+      be. Valid range is [0, $\\infty$].
```

Comment:
Make sure you set up the docs to render latex if you do this. Check the axon `mix.exs` for how